cfme/infrastructure/host.py

```diff
@@ -497,111 +422,93 @@ def untag(self, tag):
         mixins.remove_tag(tag)
 
 
-@navigator.register(Host, 'All')
+@navigator.register(Host)
 class All(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+    VIEW = HostsView
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
 
     def step(self):
         try:
-            self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+            self.prerequisite_view.navigation.select("Compute", "Infrastructure", "Hosts")
         except NoSuchElementException:
-            self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Nodes')
+            self.prerequisite_view.navigation.select("Compute", "Infrastructure", "Nodes")
 
     def resetter(self):
-        tb.select("Grid View")
-        sel.check(paginator.check_all())
-        sel.uncheck(paginator.check_all())
+        if self.view.toolbar.view_selector.selected != "Grid View":
+            self.view.toolbar.view_selector.select("Grid View")
+        self.view.paginator.check_all()
+        self.view.paginator.uncheck_all()
 
 
-@navigator.register(Host, 'Details')
+@navigator.register(Host)
 class Details(CFMENavigateStep):
-    prerequisite = NavigateToSibling('All')
+    VIEW = HostDetailsView
+    prerequisite = NavigateToSibling("All")
 
     def step(self):
-        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
-
-    def am_i_here(self):
-        return match_page(summary="{} (Summary)".format(self.obj.name))
+        self.prerequisite_view.items.get_item(by_name=self.obj.name).click()
```

Comment:
it may not be on current page. you need surf_pages=True.
btw, I added special method for such cases - https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L2101